lib/src/utils/utils.dart

```diff
@@ -27,3 +28,26 @@ Size getDefaultSize(BuildContext context) {
 double translateRotatedPosition(double size, double degree) {
   return (size / 4) * math.sin(radians(degree.abs()));
 }
+
+/// returns a dashed path from a path
+Path generateDashedPath(Path path, List<int> dashArray) {
```

Comment:
I highly recommend you to implement these kinds of functions with [extension methods](https://dart.dev/guides/language/extension-methods),
It's really really easy to use,
for example we can use 
`canvas.drawDashedLine(from, to, paint, arr)`
instead of 
`drawDashedLine(canvas, from, to, touchLinePaint, indicatorData.indicatorBelowLine.dashArray);`

and 
`path.toDashPath(arr)`
instead of 
`barPath = generateDashedPath(barPath, arr);`